Sources/Core/UnsafeMutableRawPointer.swift

```diff
@@ -0,0 +1,68 @@
+// Copyright © 2022 Max Desiatov <max@desiatov.com>.
+// All Rights Reserved.
+// SPDX-License-Identifier: BSD-3
+
+@frozen
+public struct UnsafeMutableRawPointer: _Pointer {
+  public typealias Pointee = UInt8
+
+  public let _rawValue: Builtin.RawPointer
+
+  @_transparent
+  public init(_ _rawValue: Builtin.RawPointer) {
+    self._rawValue = _rawValue
+  }
+
+  @_transparent
+  public init<T>(@_nonEphemeral _ other: UnsafeMutablePointer<T>) {
+    _rawValue = other._rawValue
+  }
+
+  @_transparent
+  public init?<T>(@_nonEphemeral _ other: UnsafeMutablePointer<T>?) {
+    guard let unwrapped = other else { return nil }
+    _rawValue = unwrapped._rawValue
+  }
+
+  @_transparent
+  public init(@_nonEphemeral mutating other: UnsafeRawPointer) {
+    _rawValue = other._rawValue
+  }
+
+  @_transparent
+  public init?(@_nonEphemeral mutating other: UnsafeRawPointer?) {
+    guard let unwrapped = other else { return nil }
+    _rawValue = unwrapped._rawValue
+  }
+
+  @inlinable
+  public static func allocate(byteCount: Int,
+                              alignment: Int) -> UnsafeMutableRawPointer {
```

Comment:
An alternate wrapping which would also work:

```suggestion
  public static func allocate(byteCount: Int, alignment: Int)
      -> UnsafeMutableRawPointer {
```